go/libraries/doltcore/merge/violations_fk_prolly.go

```diff
@@ -146,39 +146,55 @@ func prollyChildSecDiffFkConstraintViolations(
 	ctx context.Context,
 	foreignKey doltdb.ForeignKey,
 	postParent, postChild *constraintViolationsLoadedTable,
-	preChildScndryIdx prolly.Map,
+	preChildSecIdx prolly.Map,
 	receiver FKViolationReceiver) error {
 	postChildRowData := durable.ProllyMapFromIndex(postChild.RowData)
-	postChildScndryIdx := durable.ProllyMapFromIndex(postChild.IndexData)
-	parentScndryIdx := durable.ProllyMapFromIndex(postParent.IndexData)
-
-	idxDesc, _ := parentScndryIdx.Descriptors()
-	partialDesc := idxDesc.PrefixDesc(len(foreignKey.TableColumns))
-	partialKB := val.NewTupleBuilder(partialDesc)
+	postChildSecIdx := durable.ProllyMapFromIndex(postChild.IndexData)
+	parentSecIdx := durable.ProllyMapFromIndex(postParent.IndexData)
 
-	primaryKD, _ := postChildRowData.Descriptors()
-	kb := val.NewTupleBuilder(primaryKD)
+	parentSecIdxDesc, _ := parentSecIdx.Descriptors()
+	partialDesc := parentSecIdxDesc.PrefixDesc(len(foreignKey.TableColumns))
+	childPriKD, _ := postChildRowData.Descriptors()
+	childPriKB := val.NewTupleBuilder(childPriKD)
 
-	err := prolly.DiffMaps(ctx, preChildScndryIdx, postChildScndryIdx, func(ctx context.Context, diff tree.Diff) error {
+	var parentSecIdxCur *tree.Cursor
+	err := prolly.DiffMaps(ctx, preChildSecIdx, postChildSecIdx, func(ctx context.Context, diff tree.Diff) error {
 		switch diff.Type {
 		case tree.AddedDiff, tree.ModifiedDiff:
-			k, v := val.Tuple(diff.Key), val.Tuple(diff.To)
-			partialKey, hasNulls := makePartialKey(
-				partialKB,
-				foreignKey.TableColumns,
-				postChild.Index,
-				postChild.IndexSchema,
-				k,
-				v,
-				preChildScndryIdx.Pool())
-			if hasNulls {
-				return nil
+			k := val.Tuple(diff.Key)
+			// TODO: possible to skip this if there are not null constraints over entire index
+			for i := 0; i < k.Count(); i++ {
+				if k.FieldIsNull(i) {
```

Comment:
this one confuses me a bit. if a child fk has `(nil,1)` and the parent only has `(1,1)` there is no a constraint violation?

Reply:
Yep, here's a BATS test with this behavior:
https://github.com/dolthub/dolt/blob/main/integration-tests/bats/verify-constraints.bats#L100

I also tried this in MySQL and it looks like this is what they expect
```sql
mysql> create table parent (i int primary key, x int, y int, index (x,y));
Query OK, 0 rows affected (0.0246 sec)
mysql> create table child (j int primary key, x int, y int, foreign key (x,y) references parent(x,y));
Query OK, 0 rows affected (0.0207 sec)
mysql> insert into parent values (0,1,1);
Query OK, 1 row affected (0.0047 sec)
mysql> insert into child values (0,1,1);
Query OK, 1 row affected (0.0098 sec)
mysql> insert into child values (1,999,null);
Query OK, 1 row affected (0.0020 sec)
```


Some docs fwiw: https://dev.mysql.com/doc/mysql-reslimits-excerpt/8.0/en/ansi-diff-foreign-keys.html

Reply:
👍 nice research. could we still put that in an enginetest so it surfaces quickly in testing?